OpenEdXMobile/src/main/java/org/edx/mobile/module/registration/view/RegistrationOptionAutoCompleteTextView.kt

```diff
@@ -0,0 +1,76 @@
+package org.edx.mobile.module.registration.view
+
+import android.content.Context
+import android.util.AttributeSet
+import android.widget.ArrayAdapter
+import androidx.appcompat.widget.AppCompatAutoCompleteTextView
+import org.edx.mobile.R
+import org.edx.mobile.module.registration.model.RegistrationOption
+
+class RegistrationOptionAutoCompleteTextView : AppCompatAutoCompleteTextView {
+    private var adapter: ArrayAdapter<RegistrationOption>? = null
+    var selectedItem: RegistrationOption? = null
+
+    constructor(context: Context?) : super(context) {}
+    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {}
+    constructor(context: Context?, attrs: AttributeSet?, defStyle: Int) : super(context, attrs, defStyle) {}
+
+    fun hasValue(value: String?): Boolean {
+        value?.let {
+            adapter?.let {
+                return getAdapterPosition(value) >= 0
+            }
+        }
+        return false
+    }
+
+    fun select(value: String?) {
+        value?.let {
+            adapter?.let {
```

Comment:
I think there is no need adapter null check as we already check-in getAdapterPosition method.